test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -959,43 +956,55 @@ def test_cgroup_periodic_update(self):
         a = {'Resource_List.select': '1:ncpus=1:mem=500mb:host=%s' %
              self.momA, ATTR_N: name}
         j = Job(TEST_USER, attrs=a)
-        j.create_script(self.eatmem_job1)
+        j.create_script(self.eatmem_job3)
         jid = self.server.submit(j)
         a = {'job_state': 'R'}
         self.server.expect(JOB, a, jid)
         self.server.status(JOB, ATTR_o, jid)
         o = j.attributes[ATTR_o]
-        if self.paths['cpuacct']:
-            self.hostA.log_match(
-                '%s;update_job_usage: CPU usage: 0.000 secs' % jid)
-        if self.paths['memory']:
-            line = self.hostA.log_match(
-                '%s;update_job_usage: Memory usage: mem=' % jid)
-            match = re.match(r'.*mem=(\d+)kb', line[1])
+        self.tempfile.append(o)
+        # Scouring the logs for initial values takes too long
+        resc_list = ['resources_used.cput', 'resources_used.mem']
+        if self.swapctl == 'true':
+            resc_list.append('resources_used.vmem')
+        qstat = self.server.status(JOB, resc_list, id=jid)
+        cput = qstat[0]['resources_used.cput']
+        self.assertEqual(cput, '00:00:00')
+        mem = qstat[0]['resources_used.mem']
+        match = re.match(r'(\d+)kb', mem)
+        self.assertFalse(match is None)
+        usage = int(match.groups()[0])
+        self.assertTrue(usage < 20000)
+        if self.swapctl == 'true':
+            vmem = qstat[0]['resources_used.vmem']
+            match = re.match(r'(\d+)kb', vmem)
             self.assertFalse(match is None)
             usage = int(match.groups()[0])
-            self.assertTrue(usage < 400000)
-            if self.swapctl == 'true':
-                line = self.hostA.log_match(
-                    '%s;update_job_usage: Memory usage: vmem=' % jid)
-                match = re.match(r'.*vmem=(\d+)kb', line[1])
-                self.assertFalse(match is None)
-                usage = int(match.groups()[0])
-                self.assertTrue(usage < 400000)
+            self.assertTrue(usage < 20000)
         # Allow some time to pass for values to be updated
         self.logger.info('Waiting for periodic hook to update usage data.')
         time.sleep(5)
+        begin = time.time()
```

Comment:
If you are going to use this to match logs, you need to int() this.  time.time() returns a float.

Reply:
Already addressed with latest commit.